### PR TITLE
Add ArrowIOTensor with input from a pyarrow.Table

### PR DESCRIPTION
**Pull request by BryanCutler**
This adds an `ArrowIOTensor` based on a `_TableIOTensor` with multiple columns. An `ArrowIOTensor` is create with the factory function `tfio.IOTensor.from_arrow()` with a `pyarrow.Table` as input.

### CODE REVIEW

**Comment by BryanCutler** [comment]
@yongtang this is mostly working, but still a few issues to iron out and I need to make a pass to fix code style. I labelled it a WIP for now, and I should be able to finish soon.

**Comment by BryanCutler** [comment]
@yongtang I think this is ready for review. There are still a few issues, but they could be added in followups. Thanks!

File: tools/dev/Dockerfile [hunk 1]
```diff
@@ -34,7 +34,7 @@ ARG CONDA_ADD_PACKAGES=""
 
 RUN conda create -y -q -n tfio-dev python=3.6 ${CONDA_ADD_PACKAGES}
 
-ARG ARROW_VERSION=0.14.1
+ARG ARROW_VERSION=0.15.1
```
Comment (BryanCutler) [inline tools/dev/Dockerfile]: TODO change to 0.16.0 once available

**Comment by yongtang** [comment]
@BryanCutler Thanks for the PR! Overall it looks good.

I would suggest to add a several simple tests in `test_io_tensor_eager.py` as well (in addition to the already added tests in `test_arrow_eager.py`. `test_arrow_eager.py` is a much more comprehensive tests suites, though `test_arrow_eager.py` is more like a `feature comparision` for each different op types.

For example, in the past hdf5 was not possible in graph mode so we skipped the test with:
```
        pytest.param(
             "hdf5", None,
             marks=[		
                 pytest.mark.skip(reason="TODO"),		
             ],		
         ),
```

until PR #730 


(Note there are still some issues as reported by #745, though at least we have some coverage and knows which "feature" is available).

For Arrow, I think we could add a simple test in `test_io_tensor_eager.py` with one or two column only, just to show if some features are supported or not. If not supported we could just use `pytest.marks` to skip and add TODO.

The added tests in `test_arrow_eager.py` are used to serve a more through coverage.

**Comment by yongtang** [comment]
@BryanCutler Also recently I saw quite some user request to ask for support in graph mode (for both dataset and iotensor). The need for graph mode support is mostly to combine with tf.data itself to form a full pipeline. In other words, user would prefer to use:
```
dataset = tf.data.Dataset.from_tensor_slices([filename1, filename2, filename3])
dataset = dataset.map(lambda f: tfio.IOTensor.from_...())
```

This will constitutes a complete data pipeline that could be deployed in production in a infinitely continuous way (after converting from_tensor_slices to a steam source of filename entries).

I think the above is a useful scenario that will help in production usage.

While IOTensor starts with eager mode in focus, it is not very far away to support graph mode. The only requirement for graph mode, is the `dtype` which has to be known at the graph construction time (`REGISTER_OP`).

We could ask user to provide the dtype (or tf.TensorSpec) information as an additional arg when graph mode is used. HDF5IOTensor is an example where dtype/spec is provided in graph mode.

Also, inside `REGISTER_OP`, it will help shape inference if the shape information is provided explicitly. However, shape is not absolutely needed. In most cases, a rank would be suffice in graph construction time. The shape information could be passed as a tensor in graph execution time later on.

Based on the above, it is possible to only ask user for `dtype` in graph mode. And convert `shape` in `REGISTER_OP("IO>ArrowReadableRead")` from attribute to tensor input. In this way, shape is not needed by user input. The `SetShapeFn([](shape_inference::InferenceContext* c)` could use Unknown shape.

**Comment by yongtang** [comment]
@BryanCutler The graph mode support could be added later in follow up PRs I think. Though it would be good to add a TODO, or add a stub test in io_tensor_eager_test.py, and mark the test as TODO, e.g.,
```
        pytest.param(
             "arrow", None,
             marks=[		
                 pytest.mark.skip(reason="TODO"),		
             ],		
         ),
```

**Comment by yongtang** [comment]
@BryanCutler Let me know if there are any questions or I can be of any help.

File: tests/test_io_tensor_eager.py [hunk 2]
```diff
@@ -278,6 +278,24 @@ def fin():
...
+def fixture_arrow(request):
+  import pyarrow as pa # pylint: disable=import-outside-toplevel
+
+  arr1 = pa.array(list(range(100)), pa.int32())
+  arr2 = pa.array(list(range(100)), pa.int64())
+  arr3 = pa.array([x * 1.1 for x in range(100)], pa.float32())
+  table = pa.Table.from_arrays([arr1, arr2, arr3], ['a', 'b', 'c'])
+
+  args = table
+  column = 'b'
```
Comment (BryanCutler) [inline tests/test_io_tensor_eager.py]: I couldn't figure out how to parametrize the fixture, so this is only testing 1 column. That's probably good enough for now though

**Comment by BryanCutler** [comment]
@yongtang I added tests in `test_io_tensor_eager` and plan to add graph support as a followup. Please take another look when you can, thanks!

File: tests/test_io_tensor_eager.py (hunk 2 above)
Comment (yongtang) [reply tests/test_io_tensor_eager.py]: @BryanCutler There are some ways to parameterize the fixture:
https://stackoverflow.com/questions/42228895/how-to-parametrize-a-pytest-fixture/56871701#56871701

though they are quite complex. Probably not worth the effort for now as the current is good enough.

**Comment by BryanCutler** [comment]
Thanks @yongtang !  I'll be working on followups and a tutorial too.

**Pull request merged**